### ISSUE TITLE
`azurerm_stream_analytics_job`: `account_key` is not required for `authentication_mode = "Msi"`

### ISSUE THREAD

**Issue by ThreeFive-O** [hashicorp/terraform-provider-azurerm#19152]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.1

### AzureRM Provider Version

3.30

### Affected Resource(s)/Data Source(s)

azurerm_stream_analytics_job

### Terraform Configuration Files

```hcl
resource "azurerm_stream_analytics_job" "example" {
  job_storage_account {
    account_name        = azurerm_storage_account.example.name
    authentication_mode = "Msi"
  }
  ...
}
```

### Debug Output/Panic Output

```shell
│ Error: Missing required argument
│
│   on streamanalytics.job.tf line 19, in resource "azurerm_stream_analytics_job" "example":
│   19:   job_storage_account {
│
│ The argument "account_key" is required, but no definition was found.
```

### Expected Behaviour

No errors and planning succeeds

### Actual Behaviour

Planning fails

### Steps to Reproduce

terraform plan

### References

PR #19120 introduced support, but `account_key` is optional when using `authentication_mode = "Msi"`

**Comment by jiaweitao001** [comment]
PR for fixing; https://github.com/hashicorp/terraform-provider-azurerm/pull/19225

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #19225. As such, I am going to mark this issue as closed.
